src/librustdoc/clean/inline.rs

```diff
@@ -229,6 +229,7 @@ fn build_external_function(cx: &mut DocContext<'_>, did: DefId) -> clean::Functi
     let asyncness = cx.tcx.asyncness(did);
     let predicates = cx.tcx.predicates_of(did);
     let (generics, decl) = clean::enter_impl_trait(cx, |cx| {
+        // NOTE: generics need to be cleaned before the decl!
```

Comment:
As I mentioned in the commit message, some of these places may not have the ordering requirement, but just to be safe I added comments in almost all of them.

The only place I didn't add a comment is the following

```
clean/inline.rs
446:            clean::enter_impl_trait(cx, |cx| (tcx.generics_of(did), predicates).clean(cx)),
```

since it seemed like a different scenario. Let me know if you think I should add a comment there though.